r/src/compute-exec.cpp

```diff
@@ -147,6 +147,7 @@ std::shared_ptr<arrow::Table> ExecPlan_read_table(
         return prepared_plan.second->ToTable();
       });
 
+  prepared_plan.first->StopProducing();
```

Comment:
I would also wait for the production to actually have stopped, not merely ask the plan to stop.
```suggestion
  prepared_plan.first->StopProducing();
  StopIfNotOk(prepared_plan.first->finished().status());
```